### PR TITLE
Pathfinding cleanups:

### PR DESCRIPTION
**Pull request by nbougalis**
* Return std::pair instead of returning by reference
* Use std algorithms when possible
* Use auto and C++14 lambdas

@JoelKatz, @seelabs

### CODE REVIEW

File: src/ripple/app/paths/PathRequest.cpp [hunk 1]
```diff
@@ -241,12 +241,14 @@ bool PathRequest::isValid (RippleLineCache::ref crCache)
...
 */
-Json::Value PathRequest::doCreate (
+std::pair<bool, Json::Value>
+PathRequest::doCreate (
     RippleLineCache::ref& cache,
     Json::Value const& value,
     bool& valid)
 {
     Json::Value status;
+    bool valid;
```
Comment (seelabs) [inline src/ripple/app/paths/PathRequest.cpp]: This shadows the parameter `valid`

Comment (nbougalis) [inline src/ripple/app/paths/PathRequest.cpp]: The parameter was supposed to go away. I must have mis-folded the branch. Will update.

File: src/ripple/app/paths/PathRequest.cpp [hunk 2]
```diff
@@ -241,12 +241,13 @@ bool PathRequest::isValid (RippleLineCache::ref crCache)
...
-Json::Value PathRequest::doCreate (
+std::pair<bool, Json::Value>
+PathRequest::doCreate (
     RippleLineCache::ref& cache,
-    Json::Value const& value,
-    bool& valid)
+    Json::Value const& value)
 {
     Json::Value status;
+    bool valid;
```
Comment (seelabs) [inline src/ripple/app/paths/PathRequest.cpp]: Nit: Initialize this.

**Comment by seelabs** [comment]
@nbougalis Nice improvements with the uses of std algorithms. As far as I can tell, the semantics of this new code is the same as the old code.

**Comment by seelabs** [comment]
:+1: (although we should make sure the CI tests run clean - I did test unity/nonunity npm & unit tests locally)

**Comment by seelabs** [comment]
Reviewed `Small cleanups` commit. Still :+1:

File: src/ripple/app/paths/PathRequest.cpp [hunk 3]
```diff
@@ -275,7 +276,7 @@ Json::Value PathRequest::doCreate (
         }
     }
 
-    return status;
+    return { valid, status };
```
Comment (vinniefalco) [inline src/ripple/app/paths/PathRequest.cpp]: Have we lost RVO?

Comment (nbougalis) [inline src/ripple/app/paths/PathRequest.cpp]: I haven't examined the assembly, but all our compilers are good enough that they should give us close-to-RVO semantics: move `status` into the pair, and RVO the pair.

**Comment by JoelKatz** [comment]
:+1:

**Comment by vinniefalco** [comment]
I don't know how I feel about this. We're "cleaning up" code that just gave us a major migraine for the last week. But its not the big cleanup it needs, its just a light dusting. Is it wise to change this code in cosmetic ways right after we invested so many man hours testing and diagnosing it?

**Comment by JoelKatz** [comment]
I have a followup pull request:
https://github.com/ripple/rippled/pull/1469

@vinniefalco We went to a lot of effort to get that knowledge, we should get everything out of it we can. In particular, we got a new test tool so we should feel better about changing it.

**Pull request closed without merging**